tests/unit/utils/test_partition_utils.py

```diff
@@ -0,0 +1,107 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for partition utilities."""
+from pathlib import Path, PurePosixPath
+
+import pytest
+from utils.partition_utils import _has_partition, get_partition_compatible_filepath
+
+PATH_CLASSES = [Path, PurePosixPath, str]
+
+NON_PARTITION_PATHS = [
+    "/absolute/path",
+    "relative/path",
+    "",
+]
+
+PARTITION_PATHS = [
+    "(default)",
+    "(default)/",
+    "(default)/path",
+    "(partition)/path",
+]
+
+PARTITION_EXPECTED_PATHS = [
+    "default",
+    "default",
+    "default/path",
+    "partition/path",
+]
+
+# Prevent us from adding nonmatching paths for tests below.
+assert len(PARTITION_PATHS) == len(
+    PARTITION_EXPECTED_PATHS
+), "Expected partition paths and input partition paths need to match 1:1"
+
+
+@pytest.mark.parametrize(
+    ("full_path", "expected"),
+    [
+        ("some/path", False),
+        ("(default)", True),
+        ("(default)/", True),
+        ("(part)/some/path", True),
+        ("(nota)partition", False),
+    ],
+)
```

Comment:
`(not/a)partition`
`(not/a)/partition`
`(NOTA)/partition`
`(not1)/partition`

We really not allowed use number and capital case?

Reply:
Correct — we had a conversation over it at https://docs.google.com/document/d/1LeJvrUJAgyoGtbNtjQH6HTFnRMaWxO2Swq_BbvyrhP0/edit?disco=AAAAyZyrkx4

Thanks for the additional test cases! Added!

Reply:
OK then, I think upper case and auto upper case, with number, maybe more better for differentiating partitions. Given that this is a historical legacy that's all for now.